airbyte-cdk/java/airbyte-cdk/README.md

```diff
@@ -0,0 +1,58 @@
+# Developing with the Java CDK
+
+This page will walk through the process of developing with the Java CDK.
+
+## Building the CDK
+
+To build and test the Java CDK, execute the following:
+
+```sh
+./gradlew :airbyte-cdk:java:airbyte-cdk:build
+```
+
+## Bumping the declared CDK version
+
+You will need to bump this version manually whenever you are making changes to code inside the CDK.
+
+While under development, the next version number for the CDK is tracked in the file: `airbyte-cdk/java/airbyte-cdk/src/main/resources/version.properties`.
+
+If the CDK is not being modified, this file will contain the most recently published version number.
+
+## Publishing the CDK to Local Maven
+
+If your connector pins to a work-in-progress `-SNAPSHOT` version of the CDK (e.g. `0.0.1-SNAPSHOT` or `0.2.0-SNAPSHOT`), Gradle will notice this and automatically run the task to build and publish it to your MavenLocal repository before running the connector's own build and test tasks.
```

Comment:
I believe we configure the local maven repo already in gradle, so gradle tasks should already just be able to invoke .`/gradlew publishToMavenLocal`. This is a task you get by applying maven-publish plugin. 

Then for the connector to see the locally published jar, it first has to be available in the local repository (we do this by copying the results of the cdk build into a new container (the contents at /root/.m2/repository) and then the connector can just see it like a normal maven dependency. There is also an alternative workflow I have used in the past where I push snapshot jars into the cloud on pull requests ci.

But this local workflow is essentially

Build cdk in one container, invoking publishToMavenLocal task as part of this build process ->  get /root/.m2/repository folder and stage it into a connector test container -> update the version reference (either gradle.properties or somewhere else) to point at -SNAPSHOT jar -> gradle resolves snapshot jar as a normal dependency when doing your gradle run

distTar is doing something similar, but I think publishing to mavenLocal is slightly faster because you aren't paying the overhead of zipping it up and writing it to disk again.